packages/rich-text-links/bin/benchmark/get-rich-text-links.ts

```diff
@@ -0,0 +1,196 @@
+          nodeType: INLINES.EMBEDDED_ENTRY,
+          data: {
+            target: {
+              sys: {
+                id: 'author',
+                type: 'Link',
+                linkType: 'Entry',
+              },
+            },
+          },
+          content: [
+            {
+              nodeType: 'text',
+              data: {},
+              marks: [],
+              value: '',
+            },
+          ],
+        },
+        {
+          nodeType: 'text',
+          data: {},
+          marks: [],
+          value: 'was an instant classic.',
+        },
+      ],
+    },
+    {
+      nodeType: BLOCKS.PARAGRAPH,
+      data: {},
+      content: [
+        {
+          nodeType: 'text',
+          data: {},
+          marks: [],
+          value: 'The first chapter is free to read: ',
+        },
+      ],
+    },
+    {
+      nodeType: BLOCKS.EMBEDDED_ENTRY,
+      data: {
+        target: {
+          sys: {
+            id: 'chapter-1',
+            type: 'Link',
+            linkType: 'Entry',
+          },
+        },
+      },
+      content: [
+        {
+          nodeType: 'text',
+          data: {},
+          marks: [],
+          value: '',
+        },
+      ],
+    },
+    {
+      nodeType: BLOCKS.PARAGRAPH,
+      data: {},
+      content: [
+        {
+          nodeType: 'text',
+          data: {},
+          marks: [],
+          value: '',
+        },
+      ],
+    },
+    {
+      nodeType: BLOCKS.PARAGRAPH,
+      data: {},
+      content: [
+        {
+          nodeType: 'text',
+          data: {},
+          marks: [],
+          value: '',
+        },
+      ],
+    },
+    {
+      nodeType: BLOCKS.UL_LIST,
+      data: {},
+      content: [
+        {
+          nodeType: BLOCKS.LIST_ITEM,
+          data: {},
+          content: [
+            {
+              nodeType: BLOCKS.EMBEDDED_ENTRY,
+              data: {
+                target: {
+                  sys: {
+                    id: 'chapter-2',
+                    type: 'Link',
+                    linkType: 'Entry',
+                  },
+                },
+              },
+              content: [
+                {
+                  nodeType: 'text',
+                  data: {},
+                  marks: [],
+                  value: '',
+                },
+              ],
+            },
+            {
+              nodeType: BLOCKS.PARAGRAPH,
+              data: {},
+              content: [
+                {
+                  nodeType: 'text',
+                  data: {},
+                  marks: [],
+                  value: '',
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    },
+  ],
+};
+
+export default new Promise((resolve) => {
+  new Benchmark.Suite()
+    .add('getRichTextLinks', () => getRichTextEntityLinks(richTextField))
+    .on('cycle', (event: { target: string }): void => {
+      console.log(`${event.target}`);
+      resolve();
+    })
+    .run();
```

Comment:
I'm not sure how either of those would keep us from resolving in the event listener, although in part because the documentation around `Benchmark.Deferred` is very poor.